File: tests/fixtures/typescript-fail/vulnerable.ts
// Fixture: planted Semgrep p/typescript finding (eval of user input).
// Used by tests to prove the semgrep-typescript hook + CI scanner detect issues.

export function runUserCode(userInput: string): unknown {
  // Semgrep rule: javascript.lang.security.audit.eval-detected
  // eslint rule: no-eval
  return eval(userInput);
}

File: tests/fixtures/csharp-fail/Vulnerable.cs
// Fixture: planted Semgrep p/csharp finding (insecure hash algorithm: MD5).
// Used by tests to prove the semgrep-csharp hook + CI scanner detect app-code issues.
using System;
using System.Security.Cryptography;
using System.Text;

namespace Fixtures
{
    public static class Vulnerable
    {
        // Semgrep rule: csharp.lang.security.insecure-hashalgorithm.*-md5
        public static byte[] WeakHash(string input)
        {
            using var md5 = MD5.Create();
            return md5.ComputeHash(Encoding.UTF8.GetBytes(input));
        }
    }
}
